### DISCUSSION TITLE
Provide only the minimum amount of messages per page by nextjs 13 app routing

### DISCUSSION THREAD

**Discussion by StanleyLiang** [amannn/next-intl#281] [Q&A]
### Is your feature request related to a problem? Please describe.

Hi, I'm looking for a way to provide a minimum amount of messages by nextjs, and I found next-intl is capable to do 
https://next-intl-docs.vercel.app/docs/usage/messages#providing-messages

Based on the example, it looks like it's working by nextjs page-routing only.

I'm wondering whether it's working by app-routing or not. I read the document to use next-intl together with server component but my bad luck I didn't find any topics related to handle minimum amount of messages.
https://next-intl-docs.vercel.app/docs/next-13/server-components

anyway, thanks for next-intl such a easy configuration package.

### Describe the solution you'd like

none

### Describe alternatives you've considered

none

**Comment by amannn** [comment]
Hello @StanleyLiang and thank you for the kind words!

The docs on [using interactive state in translations in Server Components](https://next-intl-docs.vercel.app/docs/next-13/server-components#using-interactive-state-in-translations) have an example on how to use only the minimum amount of messages.

I'm planning to rewrite the docs to be App Router-first, hopefully this should be more visible in the future!

One note: I haven't personally experimented with assembling namespaces automatically as shown in https://next-intl-docs.vercel.app/docs/usage/messages#providing-messages. In case you're exploring this I'd be curious about your feedback!

I'll convert this to a discussion for the time being, as it's not a feature request per se.
